mindsdb/integrations/handlers/autokeras_handler/autokeras_handler.py

```diff
@@ -0,0 +1,151 @@
+from typing import Optional
+import os
+import pandas as pd
+import numpy as np
+import autokeras as ak
+from mindsdb.integrations.libs.base import BaseMLEngine
+from tensorflow.keras.models import load_model
+
+import logging
+
+def configure_logger(log_file: str = "ml_handler_logs.txt") -> None:
+    """Configure all the options for an app-wide logger.
+
+    Parameters
+    ----------
+    log_file : str
+        Path to the file where logs will be written to. File mode is set to append.
+    """
+
+    logging.basicConfig(
+        format="%(asctime)s  %(levelname)s  %(message)s",
+        filename=log_file,
+        datefmt="%Y-%m-%d %H:%M:%S",
+        filemode="a+",
+        level=logging.INFO,
+    )
+    return logging.getLogger("atlas-remote-logger")
+
+def create_logger(log_file: str = "ml_handler_logs.txt", log_name: str = "mindsdb-logger-x") -> logging.Logger:
+    """Configure and create a local logger.
+
+    Returns
+    -------
+    logger : logging.Logger
+        The pre-configured logger.
+    """
+
+    formatter = logging.Formatter("%(asctime)s  %(levelname)s  %(message)s")
+    handler = logging.FileHandler(log_file, mode="a+")
+    handler.setFormatter(formatter)
+    logger = logging.getLogger(log_name)
+    logger.setLevel(logging.DEBUG)
+    logger.addHandler(handler)
+    return logger
+
+
+logger = create_logger()
+
+
+
+# Makes this run on Windows Subsystem for Linux
+os.environ["XLA_FLAGS"]="--xla_gpu_cuda_data_dir=/usr/lib/cuda"
+
+trainer_dict = {
+    "regression": ak.StructuredDataRegressor
+}
+
+DEFAULT_MODE = "regression"
+DEFAULT_EPOCHS = 3
+DEFAULT_TRIALS = 3
+
+def train_autokeras_model(df, target, mode=DEFAULT_MODE):
+    training_df = df.drop(target, axis=1)
+    trainer = trainer_dict[mode](overwrite=True, max_trials=DEFAULT_TRIALS)
+
+    logger.info("In train(): before training")
+    logger.info(df)
+    logger.info(target)
+    logger.info(training_df)
+
+    numeric_column_names = training_df.select_dtypes(include=[np.number]).columns.values.tolist()
+
+    training_df = pd.get_dummies(training_df)
```

Comment:
This will only generate dummies from the string columns i.e. what happens to the numeric columns? Assuming they remain in the DF but just wanted to check.

Reply:
Yep, the `get_dummies()` function converts only string columns and doesn't touch the numeric ones. The return is a df with the numeric and the dummies in it.